### PR TITLE
feat(w1r3/java): report cpu ns per byte

### PR DESCRIPTION
**Pull request by BenWhitehead**
Introduce new Instrumentation class to encapsulate the pre & post execution measurment ordering.

refactoring: move latency measurement into Instrumentation along with cpu ns per byte.

### CODE REVIEW

File: w1r3/java/src/main/java/W1R3.java [hunk 1]
```diff
@@ -176,12 +177,22 @@ private void worker(
...
             .histogramBuilder("ssb/w1r3/latency")
-            .setExplicitBucketBoundariesAdvice(makeLatencyBoundaries())
+            .setExplicitBucketBoundariesAdvice(bucketBoundaries)
             .setUnit("s")
             .build();
+
+    var cpuPerByteHistogram =
+        meter
+            .histogramBuilder("ssb/w1r3/cpu")
+            .setExplicitBucketBoundariesAdvice(bucketBoundaries)
```
Comment (coryan) [inline w1r3/java/src/main/java/W1R3.java:191]: You are going to need different bucket boundaries for the CPU histogram.  Consider:

https://github.com/googleapis/storage-shared-benchmarking/blob/3a16030ea2b608bc022c71e0baa497d56bf65397/w1r3/go/main.go#L609-L622

File: w1r3/java/src/main/java/runtime/Instrumentation.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    public void report() {
+      long endNs = System.nanoTime();
+      long endCpuNs = CpuTime.getTotalCpuNanos();
+
+      long elapsed = endNs - beginNs;
+      long diffCpuNs = endCpuNs - beginCpuNs;
+
+      double latencyS = elapsed / NANOS_PER_SECOND;
+      double cpuPerByte = objectSize / ((double) diffCpuNs);
```
Comment (coryan) [inline w1r3/java/src/main/java/runtime/Instrumentation.java:66]: Shouldn't this be:

```suggestion
      double cpuPerByte = ((double) diffCpuNs) / objectSize ;
```

**Review by coryan — CHANGES_REQUESTED** [review]

**Comment by coryan** [comment]
It also looks like you need to change:

https://github.com/googleapis/storage-shared-benchmarking/blob/3a16030ea2b608bc022c71e0baa497d56bf65397/w1r3/java/Dockerfile#L25

To something like:

```
COPY src/main/java/*.java /var/tmp/build/src/main/java/
```

File: w1r3/java/src/main/java/W1R3.java (hunk 1 above)
Comment (BenWhitehead) [reply w1r3/java/src/main/java/W1R3.java:191]: I'll change it, probably want to update the design doc as well. The doc says we can use the same buckets for latency and cpu.

File: w1r3/java/src/main/java/runtime/Instrumentation.java (hunk 2 above)
Comment (BenWhitehead) [reply w1r3/java/src/main/java/runtime/Instrumentation.java]: yes, fixed

**Comment by BenWhitehead** [comment]
I also fixed the dockerfile

**Review by coryan — APPROVED** [review]

**Pull request merged**